src/main/features/claimant-response/routes/payment-option.ts

```diff
@@ -2,25 +2,146 @@ import * as express from 'express'
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.IMMEDIATELY
+        courtOfferedPaymentIntention.paymentDate = MomentFactory.currentDate().add(5,'days')
+      }
+      return courtOfferedPaymentIntention
+    }
+
+    if (decisionType === DecisionType.COURT) {
+      const paymentPlanFromDefendantFinancialStatement: PaymentPlan = PaymentPlanHelper.createPaymentPlanFromDefendantFinancialStatement(claim)
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.INSTALMENTS) {
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.INSTALMENTS
+        courtOfferedPaymentIntention.repaymentPlan = {
+          firstPaymentDate: paymentPlanFromDefendantFinancialStatement.startDate,
+          instalmentAmount: paymentPlanFromDefendantFinancialStatement.instalmentAmount,
+          paymentSchedule: Frequency.toPaymentSchedule(paymentPlanFromDefendantFinancialStatement.frequency),
+          completionDate: paymentPlanFromDefendantFinancialStatement.calculateLastPaymentDate(),
+          paymentLength: paymentPlanFromDefendantFinancialStatement.calculatePaymentLength()
+        }
+        return courtOfferedPaymentIntention
+      }
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.BY_SPECIFIED_DATE) {
+        const paymentPlanFromDefendantFinancialStatement: PaymentPlan = PaymentPlanHelper.createPaymentPlanFromDefendantFinancialStatement(claim)
+        courtOfferedPaymentIntention.paymentDate = paymentPlanFromDefendantFinancialStatement.calculateLastPaymentDate()
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.BY_SPECIFIED_DATE
+
+        return courtOfferedPaymentIntention
+      }
+    }
+
+    if (decisionType === DecisionType.DEFENDANT) {
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.BY_SPECIFIED_DATE) {
+        courtOfferedPaymentIntention.paymentDate = claimResponse.paymentIntention.paymentDate
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.BY_SPECIFIED_DATE
+      }
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.INSTALMENTS) {
+
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.INSTALMENTS
+        courtOfferedPaymentIntention.repaymentPlan = claimResponse.paymentIntention.repaymentPlan
+      }
+      return courtOfferedPaymentIntention
+    }
+  }
+
+  generateCourtCalculatedPaymentIntention (draft: Draft<DraftClaimantResponse>, claim: Claim, decisionType: DecisionType) {
+    if (decisionType === DecisionType.CLAIMANT_IN_FAVOUR_OF_DEFENDANT && draft.document.alternatePaymentMethod.paymentOption.option.value === PaymentOption.BY_SPECIFIED_DATE) {
+      return undefined
+    }
+
+    const courtCalculatedPaymentIntention = new PaymentIntention()
+    const paymentPlan: PaymentPlan = PaymentPlanHelper.createPaymentPlanFromDefendantFinancialStatement(claim)
```

Comment:
please write unit tests for `PaymentPlanHelper.createPaymentPlanFromDefendantFinancialStatement`

Reply:
done